src/TropicalGeometry/matrix.jl

```diff
@@ -30,12 +32,9 @@ julia> det(A)
 ```
 """
 function det(A::Generic.MatSpaceElem{R}) where {R<:Union{TropicalSemiringElem,MPolyRingElem{<:TropicalSemiringElem},PolyRingElem{<:TropicalSemiringElem}}}
-    detA = zero(base_ring(A))
-    nrows(A)!=ncols(A) && return detA # return tropical zero if matrix not square
-    for sigma in AbstractAlgebra.SymmetricGroup(nrows(A)) # otherwise follow Leibniz Formula
-        detA += prod([ A[i,sigma[i]] for i in 1:nrows(A) ])
-    end
-    return detA
+    T = base_ring(A)
+    nrows(A)!=ncols(A) && return zero(T) # return tropical zero if matrix not square
```

Comment:
I am curious about this choice: Why not an error? For non-tropical matrices I get an error:
```
julia> A = matrix(ZZ, [0 1 2; 3 4 5])
[0   1   2]
[3   4   5]

julia> det(A)
ERROR: Non-square matrix
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:35
 [2] det(x::ZZMatrix)
   @ Nemo ~/.julia/packages/Nemo/WUukL/src/flint/fmpz_mat.jl:749
 [3] top-level scope
   @ REPL[3]:1
```

Reply:
Suggestion accepted.